### ISSUE TITLE
Full screen option in Preferences → Display does not reflect Ctrl+F

### ISSUE THREAD

**Issue by irydacea** [wesnoth/wesnoth#6127]
When pressing the (default) Ctrl+F hotkey to toggle the Full screen option with the Preferences dialog open on any page, the Full screen checkbox in Preferences → Display does not get updated to reflect the new value of the option. This looks especially unusual if the player uses Ctrl+F so that the checkbox displays the opposite of the current value — then the checkbox needs to be toggled twice to reflect reality again.

Closing and reopening the dialog shows the option with its current value again, naturally.

Fixing this would require the UI to be notified of the player toggling the option via hotkey somehow.

This affects 1.15.x and the dev branch and has presumably affected every Wesnoth version in existence to date.

**Comment by Wedge009** [comment]
Yeah, present in 1.14 as well. I suppose it's a bit unusual that one would go to the effort of looking up the Preferences dialogue only to use the keyboard short-cut to change the windowed/full-screen display state. But it's a valid situation.

**Comment by Wedge009** [comment]
Similarly, if resizing the window manually via mouse, the resolution figures are not updated in the Preferences dialogue.

**Comment by Wedge009** [comment]
For reference:
Keyboard hot-key calling video full-screen:
https://github.com/wesnoth/wesnoth/blob/ca90694141d8e2c5d2d1ea8afcfb071e9a439f12/src/hotkey/command_executor.cpp#L629-L630

Video full-screen setting preferences variable:
https://github.com/wesnoth/wesnoth/blob/ca90694141d8e2c5d2d1ea8afcfb071e9a439f12/src/video.cpp#L558-L559

Preferences variable:
https://github.com/wesnoth/wesnoth/blob/ca90694141d8e2c5d2d1ea8afcfb071e9a439f12/src/preferences/general.cpp#L430-L433

Preferences dialogue full-screen toggle button (this has the special handling to update the dialogue, including resolution dimensions):
https://github.com/wesnoth/wesnoth/blob/ca90694141d8e2c5d2d1ea8afcfb071e9a439f12/src/gui/dialogs/preferences_dialog.cpp#L397-L405
https://github.com/wesnoth/wesnoth/blob/ca90694141d8e2c5d2d1ea8afcfb071e9a439f12/src/gui/dialogs/preferences_dialog.cpp#L1048-L1058

Not sure if it's relevant but I also saw this in the Preferences object code:
https://github.com/wesnoth/wesnoth/blob/ca90694141d8e2c5d2d1ea8afcfb071e9a439f12/src/preferences/general.cpp#L106-L133

**Comment by Wedge009** [comment]
I confirm on Windows and Linux/X11 this is no longer an issue. Please re-open someone finds it still is - and report the usual details as well, please.

**Comment by Wedge009** [comment]
Oops, I thought this was about the window size shown - that gets updated correctly. This is about the state of the full-screen toggle tick-box.
